api/firmware/query.go

```diff
@@ -70,27 +71,37 @@ func (device *Device) rawQueryV7(msg []byte) ([]byte, error) {
 		break
 	}
 
-	var lastQueryTime time.Time
-
+	// saving 11 timestamps to show the last 10 intervals.
+	lastQueryTimes := ring.New(11)
 	for {
 		switch status {
 		case hwwRspAck:
 			return payload, nil
 		case hwwRspBusy:
 			return nil, errp.New("unexpected hwwRspBusy response")
 		case hwwRspNack:
-			if lastQueryTime.IsZero() {
+			lastQueryTimes = lastQueryTimes.Prev()
+			if lastQueryTimes.Prev().Value == nil {
 				device.log.Debug("unexpected NACK response in first loop iteration")
```

Comment:
btw there is another Query() up on line 59. that should also be added to the query times.

Reply:
Right, added :+1: